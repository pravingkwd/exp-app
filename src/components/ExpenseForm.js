import React from 'react';

import 'react-dates/initialize';
import { SingleDatePicker } from 'react-dates';


import moment from 'moment';

const now = moment();


class ExpenseForm extends React.Component {

    constructor (props)
    {
        super(props);
        this.state = {
            description : props.expense ? props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense ? (props.expense.amount/100).toString() : '',
            createdAt : props.expense ? moment(props.expense.createdAt) : moment(),
            calenderFocused : false,
            error:''
        }
    }

        

    onDescriptionChange = (e) => {
        const description = e.target.value; 
        this.setState(()=> ({ description }))   
    }

    onNotechange = (e) => {
        const note = e.target.value;
        this.setState(()=>({note}));
    }

    onAmountChange = (e) => {
        const amount = e.target.value;
        const regExp = /^\d*(\.\d{0,2})?$/
        if(!amount || amount.match(regExp)) 
        {
            this.setState( () => ({ amount }))
        }
    }

    onDateChange = (createdAt) => {
        if(createdAt) {
        this.setState( () => ({ createdAt }));
        }
    }

    onFocusChange = ({focused}) => {
        this.setState( () => ({ calenderFocused : focused }) )
    }

    onSubmit = (e) => {
        e.preventDefault();
        if(!this.state.description && !this.state.amount) 
        {
          this.setState( () => ({
              error : 'Please enter description and amount'
          }) );  
        }
        else
        {
            this.setState( () => ({
                error : ''
            }) );  
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount,10) * 100,
                createdAt : this.state.createdAt.valueOf(),
                note:this.state.note
            })

        }
    }

    render(){
        return (
            <div>
            <h1>Add Expense</h1>
            { this.state.error && (<p>{this.state.error}</p>)}
            <form onSubmit={this.onSubmit}>
            <input type="text" placeholder="description" onChange={this.onDescriptionChange} value={this.state.description} />
            <input type="text" placeholder="amount" value={this.state.amount} onChange={this.onAmountChange} />
            <SingleDatePicker
            date ={this.state.createdAt}
            onDateChange={this.onDateChange} 
            focused={this.calenderFocused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={ () => false }

            />
            <textarea placeholder="add note for your expense (optional)" value={this.state.note} onChange={ this.onNotechange } ></textarea>
            <button>Add Expense </button>
            </form>
            </div>
        );


    }


}

export default ExpenseForm;