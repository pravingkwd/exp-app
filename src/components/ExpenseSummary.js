import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = ({ ExpensesCount, ExpensesTotal }) => {

    const expenseWord = ExpensesCount === 1 ? 'expense':'expenses';

    const formattedExpense = numeral(ExpensesTotal /100).format('$0,0,00');

    return (
        <div>
        <h1>Viewing {ExpensesCount} {expenseWord} totalling {formattedExpense}</h1>
        </div>
    );

}


const mapStateToProps = (state) => {

    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        ExpensesCount: visibleExpenses.length,
        ExpensesTotal: selectExpensesTotal(visibleExpenses)
    }

}

export default connect (mapStateToProps)(ExpenseSummary);



