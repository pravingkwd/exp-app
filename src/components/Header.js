import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <div>
    <h1>Expensify App </h1>
    <NavLink to="/" activeClassName="is-active">Dashboard</NavLink>
    <NavLink to="/create" activeClassName="is-active">Add Expense</NavLink>
 
    <NavLink to="/help" activeClassName="is-active">Help</NavLink>
    </div>
);

export default Header;