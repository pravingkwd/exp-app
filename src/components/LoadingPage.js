import React from 'react';

const LoadingPage = () =>{
    return (<div className="loader"><img  src="/images/loader.gif"></img></div>)
}

export default LoadingPage;