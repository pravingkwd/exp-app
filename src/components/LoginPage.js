import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

 export const LoginPage = ({startLogin}) => 
     (
        <div className="box-layout">
        <div className="box-layout__box">
        <h1>Expensify App</h1>
        <p className="box-layout__title">you can control your expenses</p>
        <button className="button" onClick={startLogin} >Login with Google</button>
        </div>
        </div>
    )

const mapDispatchToProps = (dispatch) => ({
    startLogin : () => dispatch(startLogin())
})

export default connect(undefined,mapDispatchToProps)(LoginPage);
 
