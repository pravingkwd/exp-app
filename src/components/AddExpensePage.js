import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { startAddExpense } from '../actions/expenses';


export class AddExpensePage extends React.Component {

    onSubmit = (expense) => {
        //props.dispatch(addExpense(expense));
        this.props.startAddExpense(expense)
        this.props.history.push('/');
    }

    render()
    {
        return (
            <div>This is from Add Expense Page
            <ExpenseForm onSubmit={this.onSubmit} />
    </div>
        );
    }

}

/*const AddExpensePage = (props) => ( <div>This is from Add Expense Page
    <ExpenseForm onSubmit={ (expense) => {
        //props.dispatch(addExpense(expense));
        props.onSubmit(expense)
        props.history.push('/');
    } }/>
    </div>   ); 
*/

    const mapDispatchToProps = (dispatch) => ({
        startAddExpense : (expense) => dispatch(startAddExpense(expense))
    })



    
export default connect(undefined, mapDispatchToProps )(AddExpensePage);