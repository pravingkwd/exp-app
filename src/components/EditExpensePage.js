import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { startEditExpense , startRemoveExpense } from '../actions/expenses';


export class EditExpensePage extends React.Component 
{

    onSubmit =  (expense) => {
        //this.props.dispatch(editExpense(this.props.expense.id,expense));
        this.props.startEditExpense(this.props.expense.id,expense);
        this.props.history.push('/');
     }
    
     onClick = () => {
        //this.props.dispatch(removeExpense({ id:this.props.expense.id}));
        this.props.startRemoveExpense({ id:this.props.expense.id});
        this.props.history.push('/');
    }  

    render()
    {
        return ( <div>
            <div className="page-header">
            <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
            </div>
            </div>
            <div className="content-container">
            <ExpenseForm expense={this.props.expense}  onSubmit={this.onSubmit}  />
            <button className="button button--secondary" onClick={this.onClick} >Remove Expense</button>
            </div>
           </div> )  
    }
}

/*const EditExpensePage = (props) => {  console.log(props); 
    return ( <div>
    <ExpenseForm expense={props.expense} onSubmit={ (expense) => {
       props.dispatch(editExpense(props.expense.id,expense));
       props.history.push('/');
    } } />
    <button onClick={ () => {
        props.dispatch(removeExpense({ id:props.expense.id}));
        props.history.push('/');
    } } >Remove</button>
   </div> )};
*/

const mapStateToProps = (state,props) => {

    return { expense:state.expenses.find( (expense) => {
        if(expense.id === props.match.params.id )
         return true;
    } )
}

}

const mapDispatchToProps = (dispatch,props) => ({
    startEditExpense: (id,expense) => dispatch(startEditExpense(id,expense)) ,
    startRemoveExpense : (data) => dispatch(startRemoveExpense(data))
})



export default connect(mapStateToProps,mapDispatchToProps)(EditExpensePage);