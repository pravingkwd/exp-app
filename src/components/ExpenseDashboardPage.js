import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSummary from './ExpenseSummary';

const ExpenseDashboardPage = () => { 
    
    return (
        
    <div>This is from Expense Dashboard Page
    <ExpenseSummary />
    <ExpenseListFilters/>
    <ExpenseList/>
    </div>  ); 
};


export default ExpenseDashboardPage;