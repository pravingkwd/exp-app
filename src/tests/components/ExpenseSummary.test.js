import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test("Should render Expenses Summary with 1 expense",() =>{
    const wrapper = shallow(<ExpenseSummary ExpensesCount={1} ExpensesTotal={125} />)
    expect(wrapper).toMatchSnapshot();
});

test("Should render Expense Summary with multiple expenses",()=>{
    const wrapper =  shallow(<ExpenseSummary ExpensesCount={10} ExpensesTotal={10000} />);
    expect(wrapper).toMatchSnapshot();
});
