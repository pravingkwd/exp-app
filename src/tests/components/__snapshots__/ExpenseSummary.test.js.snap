// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Expense Summary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    10
     
    expenses
     totalling 
    $100
  </h1>
</div>
`;

exports[`Should render Expenses Summary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1
  </h1>
</div>
`;
